Restructure automatic checks docs sections (#573)
canonical/sphinx-docs-starter-pack: docs/reference/github-workflows.rst

================

The primary documentation workflow checks spelling, links, and inclusive language in a documentation project; these are the same checks as described in :ref:`run-documentation-checks`. The ``documentation-checks.yaml`` workflow covers these three checks and can be added to a new or existing workflow's jobs with:

.. code:: yaml

  jobs:
    [...]
    documentation-checks:
      uses: canonical/documentation-workflows/.github/workflows/documentation-checks.yaml@main
      with:
        working-directory: 'docs'

Workflows are also available for each individual check so that projects may run a subset of
those defined in ``documentation-checks.yaml``:

.. code:: yaml
  
  jobs:
    spell-check:
      uses: canonical/documentation-workflows/.github/workflows/spelling-check.yaml@main
      with:
        working-directory: "docs"
    inclusive-language-check:
      uses: canonical/documentation-workflows/.github/workflows/inclusive-language-check.yaml@main
      with:
        working-directory: "docs"    
    link-check:
      uses: canonical/documentation-workflows/.github/workflows/link-check.yaml@main
      with:
        working-directory: "docs"

Input
-----

> The table below lists the inputs for the `documentation-checks`.yaml workflow. If your project consumes the 

Reviewer: ```suggestion
The table below lists the inputs for the ``documentation-checks.yaml`` workflow. If your project consumes the 
```

Starter Pack in a non-traditional way, declare any of the following inputs to customize the workflow as needed:

.. list-table::
   :header-rows: 1

   * - Input
     - Description
     - Default
   * - ``working-directory``
     - The root of the documentation project. This input is required.
     - None
   * - ``python-version``
     - The Python interpreter to use for the workflow's jobs.

>      - ``'3.10'``

Reviewer: ```suggestion
     - The default Python version used. Example: ``'3.10'``
```

See comment re Ubuntu version below.

   * - ``fetch-depth``
     - The number of commits to fetch from your repository.
     - The full history is fetched.
   * - ``runs-on``
     - The host system for the workflow's runners.

>      - ``'["ubuntu-24.04"]'``

Reviewer: ```suggestion
     - The current Ubuntu LTS. Example: ``'["ubuntu-24.04"]'``
```

Not necessary, but phrasing it as an example means you wouldn't need to update this document every time the default changes.

   * - ``makefile``
     - The Makefile that checks are invoked from.
     - ``'Makefile'``
   * - ``install-target``
     - The make target for installing required tools.
     - ``'install'``
   * - ``spelling-target``
     - The make target to run for the spelling check.
     - ``'spelling'``
   * - ``woke-target``
     - The make target to run for the inclusive language check.
     - ``'woke'``
   * - ``linkcheck-target``
     - The make target to run for the link check.
     - ``'linkcheck'``

Check for removed URLs
----------------------

.. versionadded:: 1.2.0

> The starter pack includes a GitHub action to identify when page URLs have been removed. This flags pages that have moved or been removed entirely.

Reviewer: ```suggestion
The Starter Pack includes a GitHub action to identify when page URLs have been removed. This flags pages that have moved or been removed entirely.
```

This does not cover higher-level changes to URL paths, such as changes to the project name or URL slug pattern on RTD.

> This check ensures that redirects are implemented when pages are moved or appropriate information is provided when something is removed.

Reviewer: ```suggestion
This check ensures that redirects are implemented when pages are moved or that appropriate information is provided when something is removed.
```

.. _github-workflows:

> GitHub Workflows

Reviewer 2: ```suggestion
GitHub workflows
```
Unlike GitHub Actions, workflows aren't treated as a proper noun in [the Github docs](https://docs.github.com/en/actions/concepts/workflows-and-actions/workflows).

============

> The primary documentation workflow checks spelling, links, and inclusive language in a documentation project. `documentation-checks.yaml` workflow covers these three checks and can be added to a new or existing workflow's jobs with:

Reviewer 2: Required: I think it's important to highlight that these checks are the same as the local checks.

Reviewer 3: This is still a bit confusing to me. Is the default behavior that all the automatic checks are invoked directly from the Makefile? (thus guaranteeing that the checks invoked by the action are exactly the same as what you'd run locally with `make <check name>`?) - if so, that should definitely be made clear.

As a follow-up: under what circumstances would someone's doc checks invoke the [old inclusive language check (woke)](https://github.com/canonical/documentation-workflows/blob/main/inclusive-language/inclusive_language.py) or the [old spellcheck (aspell)](https://github.com/canonical/documentation-workflows/blob/main/spellcheck/spellcheck.py)?

Reviewer 2: > Is the default behavior that all the automatic checks are invoked directly from the Makefile?

Despite the fact that woke and aspell are installed, the workflows don't invoke them at any point (unless the caller's Makefile still does). The most important lines in the workflows are [these ones](https://github.com/canonical/documentation-workflows/blob/main/inclusive-language/inclusive_language.py#L35-L36). AFAIK, these are only invoked if a team still has the old `Makefile.sp` file.

It could be that their mere presence somehow messes things up, but I'm not sure why that would be.

.. code:: yaml

  jobs:
    [...]
    documentation-checks:
      uses: canonical/documentation-workflows/.github/workflows/documentation-checks.yaml@main
      with:
        working-directory: 'docs'

Workflows are also available for each individual check so that projects may run a subset of
those defined in ``documentation-checks.yaml``:

.. code:: yaml
  
  jobs:
    spell-check:
      uses: canonical/documentation-workflows/.github/workflows/spelling-check.yaml@main
      with:
        working-directory: "docs"
    inclusive-language-check:
      uses: canonical/documentation-workflows/.github/workflows/inclusive-language-check.yaml@main
      with:
        working-directory: "docs"    
    link-check:
      uses: canonical/documentation-workflows/.github/workflows/link-check.yaml@main
      with:
        working-directory: "docs"

Input
-----

> The table below lists the inputs for the various CI workflows. If your project consumes the 

Reviewer 2: ```suggestion
The table below lists the inputs for the `documentation-checks`.yaml workflow. If your project consumes the 
```
This is a nit, but the individual workflows each support a subset of these. I wouldn't worry about pointing this out since they're only for edge cases.

   * - Input
     - Description
     - Default
   * - ``working-directory``
     - The root of the documentation project. This input is required.
     - None
   * - ``python-version``
     - The Python interpreter to use for the workflow's jobs.
     - ``'3.10'``
   * - ``fetch-depth``
     - The number of commits to fetch from your repository.
     - The full history is fetched.
   * - ``runs-on``
     - The host system for the workflow's runners.
     - ``'["ubuntu-24.04"]'``
   * - ``makefile``
     - The Makefile that checks are invoked from.
     - ``'Makefile'``
   * - ``install-target``
     - The make target for installing required tools.
     - ``'install'``
   * - ``spelling-target``
     - The make target to run for the spelling check.
     - ``'spelling'``
   * - ``woke-target``
     - The make target to run for the inclusive language check.
     - ``'woke'``
   * - ``linkcheck-target``
     - The make target to run for the link check.
     - ``'linkcheck'``

Check for removed URLs
----------------------

.. versionadded:: 1.2.0

The starter pack includes a GitHub action to
identify when page URLs have been removed. This includes moving pages to another

> path, or removing them completely.

Reviewer 2: Optional:
```suggestion
identify when page URLs have been removed. This flags pages that have moved or been removed entirely.
```

This does not cover higher-level changes to URL paths, such as changing the RTD 

> project name, or language and versioning structure provided by RTD.

Reviewer 2: Optional:
```suggestion
This does not cover higher-level changes to URL paths, such as changes to the 
project name or URL slug patterns on RTD.
```

This check is available to ensure that redirects are implemented when pages are

> moved, or appropriate information can be provided when anything is removed.

Reviewer 2: Optional
```suggestion
This check ensures that redirects are implemented when pages are
moved or appropriate information is when something is removed.
```
